newlib/libc/sys/vita/pipe.c

```diff
@@ -0,0 +1,77 @@
+/*
+
+Copyright (C) 2022, vitasdk
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+
+*/
+
+#include <reent.h>
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/syslimits.h>
+#include <string.h>
+
+#include <psp2/types.h>
+#include "vitaerror.h"
+#include "vitadescriptor.h"
+
+#define MSGPIPE_MEMTYPE_USER_MAIN 0x40
+#define MSGPIPE_THREAD_ATTR_PRIO (0x8 | 0x4)
+
+
+int pipe(int pipefd[2])
+{
+    int ret;
+    ret = sceKernelCreateMsgPipe("newlib pipe", MSGPIPE_MEMTYPE_USER_MAIN, MSGPIPE_THREAD_ATTR_PRIO, 4096, NULL);
```

Comment:
do we need to make buffer size value as `weakref` to override them or how about give large buffer for them?
bsds support [16K~64K pipe buffers](https://www.netmeister.org/blog/ipcbufs.html).
(but giving large value would make another problems.)

Reply:
No. 4096 is max

Reply:
is it limitation of sceKernelCreateMsgPipe?

Reply:
Yes

Reply:
can you recheck of this? maybe we have to change `vita-header` part.
document said ` * @param bufSize - The size of the internal buffer in multiples of 0x1000 (4KB)`

Reply:
hm. you're right. it's 32Mb max per process, with 4kb align.
Should i bump it up to 16k then?

Reply:
yep. setting large value could be good for applications